widget.subscribe.php

```diff
@@ -359,33 +359,8 @@ function add_subscriber() {
 				}
 			} //end for loop
 
-
-			$subscribe_to_lists  = [];
-			if ( isset($_POST['sailthru_email_list']) ){
-				$sailthru_email_list = sanitize_text_field( $_POST['sailthru_email_list'] );
-			}
-			if ( ! empty( $sailthru_email_list ) ) {
-
-				// check for double opt in setting
-				if ( isset( $customfields['sailthru_double_opt_in'] ) && true === $customfields['sailthru_double_opt_in'] ) {
-					$double_opt_in = true;
-				} else {
-					$double_opt_in = false;
-				}
-
-				$lists = explode( ',', $sailthru_email_list );
-
-				foreach ( $lists as $key => $list ) {
-					$subscribe_to_lists[ $list ] = 1;
-				}
-
-				$options['lists'] = $subscribe_to_lists;
-
-			} else {
-
-				$options['lists'] = array( 'Sailthru Subscribe Widget' => 1 ); // subscriber is an orphan
-
-			}
+            $double_opt_in = !empty( $customfields['sailthru_double_opt_in'] );
+            $subscribe_to_lists = $this->extract_subscribe_to_list();
```

Comment:
Both of these could be moved much closer to where they are actually being used (down to around line 440)

Reply:
Yeah, I agree 100% with that and implemented like that first, but then I moved back to the original spot to make it clear in review what the new code was replacing. I'll move them to the place where they are used.

Reply:
Sounds good